odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+                axis))
+
+        return axis
+
+    def det_rotation(self, angle):
+        """The detector rotation function.
+
+        Returns the matrix for rotating a vector in 3d by an angle ``angle``
+        about the rotation axis given by the property `axis` according to
+        the right hand rule.
+
+        The matrix is computed according to
+        `Rodrigues' rotation formula
+        <https://en.wikipedia.org/wiki/Rodrigues'_rotation_formula>`_.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's `motion_params`.
+
+        Returns
+        -------
+        rot_mat : `numpy.matrix`, shape ``(3, 3)``
+            The rotation matrix mapping the standard basis vectors in
+            the fixed ("lab") coordinate system to the basis vectors of
+            the local coordinate system of the detector reference point,
+            expressed in the fixed system.
+        """
+
+        angle = float(angle)
+        if angle not in self.motion_params:
+            raise ValueError('angle {} is not in the valid range {}.'
+                             ''.format(angle, self.motion_params))
+
+        axis = self.axis
+
+        cross_mat = np.matrix([[0, -axis[2], axis[1]],
+                               [axis[2], 0, -axis[0]],
+                               [-axis[1], axis[0], 0]])
+        dy_mat = np.asmatrix(np.outer(axis, axis))
+        id_mat = np.asmatrix(np.eye(3))
+        cos_ang = cos(angle)
+        sin_ang = sin(angle)
+
+        return cos_ang * id_mat + (1. - cos_ang) * dy_mat + sin_ang * cross_mat
+
+        # return euler_matrix(angle)
+
+    def __repr__(self):
+        """`g.__repr__() <==> repr(g)`."""
+        inner_fstr = '{!r}, {!r}, src_rad={}, det_rad={}'
+        if self.has_motion_sampling:
+            inner_fstr += ',\n agrid={agrid!r}'
+        if self.has_det_sampling:
+            inner_fstr += ',\n dgrid={dgrid!r}'
+        inner_str = inner_fstr.format(self.motion_params, self.det_params,
+                                      self.src_radius, self.det_radius,
+                                      agrid=self.motion_grid,
+                                      dgrid=self.det_grid)
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+    def __str__(self):
+        """`g.__str__() <==> str(g)`."""
+        return self.__repr__()  # TODO: prettify
+
+
+class ConeFlatGeometry(ConeBeamGeometry):
+    """Cone beam geometry in 3d with flat detector.
+
+    The source moves on a circle with radius ``r``, and the detector
+    reference point is opposite to the source on a circle with radius ``R``
+    and aligned tangential to the circle.
+
+    The motion parameter is the (1d) rotation angle parametrizing source and
+    detector positions.
+    """
+
+    def __init__(self, angle_intvl, dparams, src_rad, det_rad, agrid=None,
```

Comment:
Agree that we should avoid an explosion of classes, but we need to decide on how to handle the discretization here.